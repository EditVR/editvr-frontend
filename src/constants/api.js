/**
 * @file user.js
 * Exports constants pertaining to API paths.
 */

export const API_ENDPOINT_USER_LOGIN = 'oauth/token';
export const API_ENDPOINT_XCSRF_TOKEN = 'rest/session/token';
export const API_ENDPOINT_EXPERIENCE = 'node/experience';
export const API_ENDPOINT_FILE_IMAGE = 'file/image';
export const API_ENDPOINT_FILE_VIDEO = 'file/video';
export const API_ENDPOINT_FILE_AUDIO = 'file/audio';
export const API_ENDPOINT_SCENE = 'node/scene';
export const API_ENDPOINT_COMPONENT = 'node/component';
export const API_TYPE_EXPERIENCE = 'node--experience';
export const API_TYPE_SCENE = 'node--scene';
export const API_TYPE_COMPONENT = 'node--component';
export const API_TYPE_FILE_IMAGE = 'file--image';
export const API_TYPE_FILE_VIDEO = 'file--video';
export const API_TYPE_FILE_AUDIO = 'file--audio';
