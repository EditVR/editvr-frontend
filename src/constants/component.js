/**
 * @file component.js
 * Exports component-related constants.
 */

export const COMPONENT_SELECT = 'COMPONENT_SELECT';
export const COMPONENT_DELETE = 'COMPONENT_DELETE';
export const COMPONENT_TYPE_DIALOG = 'panelimage';
export const COMPONENT_TYPE_LINK = 'link';
