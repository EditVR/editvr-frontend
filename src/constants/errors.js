/**
 * @file errors.js
 * Exports error constants.
 */

export const ERROR_API_CONNECTION_FAILED = 'Unable to connect to EditVR API';
export const ERROR_API_FORBIDDEN =
  'Access to perform this action has been denied.';
export const ERROR_API_NOT_FOUND = 'Unable to find API resource.';
export const ERROR_API_BAD_REQUEST = 'Input information is invalid.';
export const ERROR_API_UNPROCESSABLE_ENTITY =
  'Unable to operate on this item, it is malformed.';
export const ERROR_API_GENERAL = 'The server encountered an error.';
export const ERROR_API_LOGIN_FAILED =
  'Unable to log in with the provided credentials.';
export const ERROR_API_REGISTER_FAILED_EMAIL =
  'Unable to register for an account with the provided email. Make sure the email is valid and not already in use.';
export const ERROR_API_REGISTER_FAILED_USERNAME =
  'Unable to register for an account with the provided username. Make sure the username is valid and not already in use.';

export const AXIOS_ERROR_NETWORK = 'Error: Network Error';
export const AXIOS_ERROR_400 = 'Error: Request failed with status code 400';
export const AXIOS_ERROR_401 = 'Error: Request failed with status code 401';
export const AXIOS_ERROR_403 = 'Error: Request failed with status code 403';
export const AXIOS_ERROR_404 = 'Error: Request failed with status code 404';
export const AXIOS_ERROR_422 = 'Error: Request failed with status code 422';
