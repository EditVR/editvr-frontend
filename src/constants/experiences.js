/**
 * @file experiences.js
 * Exports constants pertaining to experiences.
 */

export const EXPERIENCES_FETCH_FOR_USER = 'EXPERIENCES_FETCH_FOR_USER';
export const EXPERIENCES_CREATE = 'EXPERIENCES_CREATE';
export const EXPERIENCES_EDIT = 'EXPERIENCES_EDIT';
export const EXPERIENCES_DELETE = 'EXPERIENCES_DELETE';
