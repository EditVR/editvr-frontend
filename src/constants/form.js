/**
 * @file form.js
 * Exports constants pertaining to forms.
 */

export const FORM_BUTTON_INSERT_UPDATE = 'Save';
export const FORM_BUTTON_REGISTER = 'Register';
export const FORM_BUTTON_LOGIN = 'Login';
export const FORM_INPUT_VALIDATION_REGEX_URL_SEGMENT = /^[a-zA-Z0-9_-]*$/;
