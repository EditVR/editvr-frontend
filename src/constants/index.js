/**
 * @file index.js
 * Exports all constants.
 */

export * from './user';
export * from './api';
export * from './errors';
export * from './modes';
export * from './form';
export * from './experiences';
export * from './openExperience';
