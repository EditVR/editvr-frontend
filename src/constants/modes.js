/**
 * @file modes.js
 * Exports constants pertaining to modes. These should be url-friendly.
 */

export const MODE_SCENE_CREATE = 'mode-scene-create';
export const MODE_SCENE_EDIT = 'mode-scene-edit';
export const MODE_COMPONENT_SELECTING = 'mode-component-selecting';
