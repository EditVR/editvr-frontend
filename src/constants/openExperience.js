/**
 * @file openExperience.js
 * Exports constants pertaining to open experiences.
 */

export const OPEN_EXPERIENCE_FETCH_FOR_USER = 'OPEN_EXPERIENCE_FETCH_FOR_USER';
export const OPEN_EXPERIENCE_SCENE_CREATE = 'OPEN_EXPERIENCE_SCENE_CREATE';
export const OPEN_EXPERIENCE_SCENE_EDIT = 'OPEN_EXPERIENCE_SCENE_EDIT';
export const OPEN_EXPERIENCE_SCENE_FIELD_PRESAVE =
  'OPEN_EXPERIENCE_SCENE_FIELD_PRESAVE';
export const OPEN_EXPERIENCE_COMPONENT_FIELD_PRESAVE =
  'OPEN_EXPERIENCE_COMPONENT_FIELD_PRESAVE';
export const OPEN_EXPERIENCE_COMPONENT_EDIT = 'OPEN_EXPERIENCE_COMPONENT_EDIT';
export const OPEN_EXPERIENCE_COMPONENT_CREATE =
  'OPEN_EXPERIENCE_COMPONENT_CREATE';
