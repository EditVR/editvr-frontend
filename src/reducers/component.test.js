/**
 * @file component.test.js
 * Contains tests for component.js.
 */

import reducer from './component';
import { COMPONENT_SELECT } from '../constants';

describe('reducers->component', () => {
  it(`Should handle ${COMPONENT_SELECT}_SUCCESS`, () => {
    expect(
      reducer(undefined, {
        type: `${COMPONENT_SELECT}_SUCCESS`,
        payload: {
          id: 'test'
        }
      })
    ).toMatchSnapshot();
  });
});
