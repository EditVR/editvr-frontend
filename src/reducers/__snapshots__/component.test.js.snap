// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers->component Should handle COMPONENT_SELECT_SUCCESS 1`] = `
Object {
  "id": "test",
}
`;
